src/openhab.cpp

```diff
@@ -70,100 +66,161 @@ OpenHAB::OpenHAB(const QVariantMap& config, EntitiesInterface* entities, Notific
     } else if (!_url.endsWith('/')) {
         _url += "/";
     }
-    _sseReconnectTimer = new QTimer(this);
-
-
-
-    qCDebug(m_logCategory) << "setup";
+    context_openHab = this;
+    _sseNetworkManager = new QNetworkAccessManager(context_openHab);
+    _sseReconnectTimer = new QTimer(context_openHab);
+    _nam = new QNetworkAccessManager(context_openHab);
+    _nam = new QNetworkAccessManager(context_openHab);
+
+    for (QNetworkInterface& iface : QNetworkInterface::allInterfaces()) {
+        if (iface.type() == QNetworkInterface::Wifi) {
+            qCDebug(m_logCategory) << iface.humanReadableName() << "(" << iface.name() << ")"
+                                   << "is up:" << iface.flags().testFlag(QNetworkInterface::IsUp)
+                                   << "is running:" << iface.flags().testFlag(QNetworkInterface::IsRunning);
+            _iface = iface;
+        }
+    }
 }
 
 void OpenHAB::streamReceived() {
-    QByteArray     rawData = _sseReply->readAll();
-    QByteArrayList splitted = rawData.split('\n');
+    if (_sseReply->error() == QNetworkReply::NoError) {
+        QJsonParseError parseerror;
 
-    for (QByteArray data : splitted) {
-        if ((data.length() == 0) || (data.startsWith("event: message"))) {
-            continue;
-        }
+        rawData.clear();
+        rawData = _sseReply->readAll();
+        splitted = rawData.split('\n');
 
-        data = data.replace("\"{", "{").replace("}\"", "}").replace("\\\"", "\"");
+        for (QByteArray data : splitted) {
+            if ((data.length() == 0) || (data.startsWith("event: message"))) {
+                continue;
+            }
 
-        QJsonParseError parseerror;
-        // remove the message start passage, because it is no valid JSON
-        QJsonDocument doc = QJsonDocument::fromJson(data.mid(6), &parseerror);
-        if (parseerror.error != QJsonParseError::NoError) {
-            // qCDebug(m_logCategory) << QString(doc.toJson(QJsonDocument::Compact));
-            // qCDebug(m_logCategory) << "read " << rawData.size() << "bytes";
-            qCCritical(m_logCategory) << "SSE JSON error:" << parseerror.errorString();
-            continue;
-        }
+            if (_flagMoreDataNeeded) {
+                _tempJSONData = _tempJSONData + QString(data);
+                doc = QJsonDocument::fromJson(_tempJSONData.toUtf8(), &parseerror);
```

Comment:
Use a local variable for doc. Doesn't seem to be used outside this method.

Reply:
done